shared/reducers/fs.js

```diff
@@ -155,6 +167,44 @@ export default function(state: Types.State = initialState, action: FsGen.Actions
         ]
       }, [])
       return state.mergeIn(['pathItems'], resetsToMerge)
+    case FsGen.newFolderRow:
+      const {parentPath} = action.payload
+      const parentPathItem = state.pathItems.get(parentPath, Constants.makeUnknownPathItem())
+      if (parentPathItem.type !== 'folder') {
+        console.warn(`bad parentPath: ${Types.pathToString(parentPath)}`)
+        return state
+      }
+      let newFolderName = 'New Folder'
+      for (let i = 2; parentPathItem.children.has(newFolderName); ++i) {
+        newFolderName = `New Folder ${i}`
+      }
+
+      return state.mergeIn(
+        ['edits'],
+        [
+          [
+            Constants.makeEditID(),
+            Constants.makeNewFolder({
+              name: newFolderName,
+              hint: newFolderName,
+              parentPath,
+            }),
+          ],
+        ]
+      )
+    case FsGen.newFolderName:
+      return state.updateIn(
+        // $FlowFixMe
```

Comment:
what is flow saying here and below

Reply:
Flow says:

```
Cannot call state.updateIn because EditID [1] is incompatible with key set [2] in type argument K2.

     reducers/fs.js
      193│         ]
      194│       )
      195│     case FsGen.newFolderName:
      196│       return state.updateIn(
      197│         ['edits', action.payload.editID],
      198│         editItem => editItem && editItem.set('name', action.payload.name)
      199│       )
      200│     case FsGen.editSuccess:
      201│     case FsGen.discardEdit:
      202│       // $FlowFixMe

     actions/fs-gen.js
 [1]  123│   editID: Types.EditID,

     node_modules/immutable/dist/immutable.js.flow
 [2] 1722│   updateIn<K: $Keys<T>, K2: $KeyOf<$ValOf<T, K>>, S: $ValOf<$ValOf<T, K>, K2>>(keyPath: [K, K2], updater: (value: $ValOf<$ValOf<T, K>, K2>) => S): this & T;
```